engine/src/main/java/io/camunda/zeebe/engine/state/compensation/DbCompensationSubscriptionState.java

```diff
@@ -75,6 +78,44 @@ public Set<CompensationSubscription> findSubscriptionsByProcessInstanceKey(
     return completedActivities;
   }
 
+  @Override
+  public Optional<CompensationSubscription> findSubscriptionByCompensationHandlerId(
+      final String tenantId, final long piKey, final String compensationHandlerId) {
+    tenantIdKey.wrapString(tenantId);
+    processInstanceKey.wrapLong(piKey);
+
+    final List<CompensationSubscription> compensationSubscription = new ArrayList<>();
+    compensationSubscriptionColumnFamily.whileEqualPrefix(
+        new DbCompositeKey<>(tenantIdKey, processInstanceKey),
+        ((key, value) -> {
+          if (value.getRecord().getCompensationHandlerId().equals(compensationHandlerId)) {
+            compensationSubscription.add(value.copy());
+          }
+        }));
+
+    if (compensationSubscription.isEmpty()) {
+      return Optional.empty();
+    }
+    return compensationSubscription.stream().findFirst();
```

Comment:
:wrench: This check is not necessary.

```suggestion
    return compensationSubscription.stream().findFirst();
```